chore(inventory): skip tags where possible

jniles opened the pull request:
This commit ensures that the `bhInventorySelect` doesn't trigger the tag search path, since it isn't needed.

To test:
On `master`
1. Build the default test database via the integration tests.
2. Add "DEBUG=db*" to the .env file
2. Navigate to the "stock sheet" report and notice that a lot of tags are queried on initial load.

On this branch:
1. Repeat the steps above.
2. Notice that the tags are not queried.

jniles committed:
chore(inventory): skip tags where possible

This commit ensures that the `bhInventorySelect` doesn't trigger the tag
search path, since it isn't needed.

jmcameron reviewed (commented):
This fix definitely speeds things up.  Nice!  However, I still see 'LEFT JOIN tags...' when going to to the Stock Sheet page --- which I presume is not necessary.

jmcameron commented on client/src/js/components/bhInventorySelect.js:
```diff
@@ -24,9 +24,11 @@ function InventorySelectController(Inventory, Notify) {
-    const params = $ctrl.onlyConsumable
-      ? { consumable : 1 }
-      : {};
+    const params = { skipTags : 1 };
```
Since you are using skipTags as a boolean, why use '1' instead of 'true'?

jniles replied:
Because of type coercion and javascript being a terrible language to work in.  Try this:

```js
// as strings of numbers
'1' == true // => true
'0' == false // => false

// as strings themselves
'true' == true // => false
'false' == false // =>true
```

Basically, even if the number are passed around as strings, they true/false properties are retained.  If "true" and "false" are passed around as strings, it changes the `if (bool) { }` behavior.

... that being said, it's much more readable to use booleans, so I'll change it and figure out where we have buggy code instead.

jmcameron reviewed (commented):
It looks pretty good.  See my note in the code. 

But I think you need to review all uses of Stock.inventories.read() to make sure that skipTags is used appropriately.  For instance, I think skipTags=true is missing for creating purchase orders.

jniles committed:
chore(inventory): update all skipTags instances

Updates the syntax of `skipTags` to be `true` instead of `1`.  Ensures
that all places that need to skip tags have the ability to skip tags.

jmcameron commented on client/src/i18n/en/form.json:
```diff
@@ -802,7 +802,7 @@
             "TYPE": "Type",
             "UNCONFIGURED": "Unconfigured",
-            "UNCONSUMABLE": "Non-consummable",
+            "UNCONSUMABLE": "Non-consumable",
```
!

jmcameron reviewed (approved):
I looked over the code changes and it looks good.  I did have time to test each of the user interfaces that could have been impacted by this change.   So I think this is ready, however, if you have not verified that the potentially affected user interfaces still work correctly, you may want to do that.
